### DISCUSSION TITLE
Use `W` as alias for `w` in command mode

### DISCUSSION THREAD

**Discussion by nuke-web3** [helix-editor/helix#8286] [General]
While waiting for a solution to create custom commands and/or command _aliases_ https://github.com/helix-editor/helix/issues/4423#issuecomment-1520599773 I propose that W (unused) would be an alias for w in the command pallet.

FAR too often I type `:W` (shift held down) instead of `:w`. So much I am considering dropping using the editor as I need to first save to do other tasks/run commands that need the file saved.

**Comment by nuke-web3** [comment]
Same could be for `Q` and `q` too, as all too often I get the same issue for the same reason

**Comment by David-Else** [comment]
Not keen on the idea myself, perhaps try:
```
[keys.normal.space]
w = ":write"
```
I played around with some options for a while and settled on the above as the best, very fast.

**Comment by nuke-web3** [comment]
That is a good suggestion, thanks!

Why do you _not_ want the mapping (at least optionally) for the command mode itself? Using it for something else?

**Comment by David-Else** [comment]
> Why do you _not_ want the mapping (at least optionally) for the command mode itself? Using it for something else?

Every key is precious, and `w` `W` seems redundant... just practice touch typing a bit more :) I use https://github.com/lemnos/tt

In Helix the capital version of keys always have another related command.

**Reply by nuke-web3** [reply]
well, in many modes, the control commands are aliased, like https://docs.helix-editor.com/keymap.html#window-mode uses many `key` and `C-key` to be aliased, not sure why `<shift>` should be thought of differently here, I presume that alias is for the exact same reason as why I want this alias.

**Comment by pascalkuthe** [comment]
I agree I don't think an alias is necessary here.

**Comment by aspizu** [comment]
I use `Ctrl+S` because it works even in insert mode.
and also `Ctrl+Q` -> `:bc`

**Reply by pascalkuthe** [reply]
saving in insertmode does not work and will lead to crashes

**Reply by aspizu** [reply]
it seems to work fine?

**Reply by pascalkuthe** [reply]
its a known issue: https://github.com/helix-editor/helix/issues/3501

The code is written under the assumption that you won't save from insert mode. If you do regardless you may encounter all forms of wed behaviour from broken undo to rashes. It may work in some scenarios but its just not properly supported

**Reply by David-Else** [reply]
Why not remove saving in insert and stop future dev pain?

**Reply by pascalkuthe** [reply]
I would be open to acceping a PR that makes commands like writ and undo just print an error in insertmode. That is pretty easy to implement and would avoid crashing. Not sure what the taught of @the-mikedavis  and @archseer on that ar

**Comment by mitjafelicijan** [comment]
Why not have the ability to have custom commands, like there are custom keybindings that override existing commands?

That would allow us to have `:W` for saving and even if this overrides something that is native, that would be our choice to do so, and we would need to live with this.

**Reply by kirawi** [reply]
See the linked issue in the parent comment.
